src/Symfony/Bridge/Doctrine/Middleware/ConnectionKeepAliveDriver.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Middleware;
+
+use Doctrine\DBAL\Driver;
+use Doctrine\DBAL\Driver\Connection as DriverConnection;
+use Doctrine\DBAL\Driver\Middleware\AbstractDriverMiddleware;
+
+final class ConnectionKeepAliveDriver extends AbstractDriverMiddleware
```

Comment:
I think the naming is off here as well… this driver does not keep connections alive, it works with another class to close them, by recording when the underlying connection was last used.

```suggestion
final class UsageRecordingDriver extends AbstractDriverMiddleware
```